src/libopensc/card-itacns.c

```diff
@@ -82,13 +85,22 @@ static const struct sc_atr_table itacns_atrs[] = {
 		idx ++; \
 	} while(0);
 
+/* Assigns the index to v (must be an address) and increment index */
+#define itacns_atr_safe_assign(idx, v, mask) do { \
+		itacns_atr_l(idx); \
+		*v = (atr[idx] & mask); \
+		idx++; \
+	} while(0);
```

Comment:
In my first review I've pointed out that the *existing* macros are unreadable and they hide an essential safety operation, i.e. length checking. You are responding with a **new** unreadable macro that hides length checking. I'm not sure what to make of this. Maybe we need a second opinion...

Reply:
I agree that's not really the best option, I just followed the style to be _at least_ consistent and not rewrite the whole thing.

But let me see a bit and find a way to achieve both goals